frontend/rolecall/src/app/user/user-editor.component.ts

```diff
@@ -1,121 +1,197 @@
+  constructor(
+      private readonly route: ActivatedRoute,
+      private readonly userAPI: UserApi,
+      private readonly location: Location) {
+  }
 
   ngOnInit(): void {
-    let uuid = this.route.snapshot.params.uuid;
-    if (!isNullOrUndefined(uuid)) {
+    const uuid: string = this.route.snapshot.params.uuid;
+    if (uuid) {
       this.urlPointingUUID = uuid;
     }
-    this.userAPI.userEmitter.subscribe((val) => { this.onUserLoad(val) });
+    this.userAPI.userEmitter.subscribe((user) => {
+      this.onUserLoad(user);
+    });
     this.userAPI.getAllUsers();
   }
 
-  onUserLoad(users: User[]) {
-    if (users.length == 0) {
+  private onUserLoad(users: User[]) {
+    if (users.length === 0) {
       this.renderingUsers = [];
       return;
     }
+
     if (this.renderingUsers) {
-      let prevUserUUIDS = new Set(this.renderingUsers.map(user => user.uuid));
-      let newUsers = [];
-      for (let user of users) {
+      const prevUserUUIDS = new Set(this.renderingUsers.map(user => user.uuid));
+      const newUsers = [];
+      for (const user of users) {
         if (!prevUserUUIDS.has(user.uuid)) {
           newUsers.push(user);
         }
       }
+
       if (newUsers.length > 0) {
-        for (let newUser of newUsers) {
-          if (newUser.contact_info.email == this.lastSelectedUserEmail) {
+        for (const newUser of newUsers) {
+          if (newUser.contact_info.email === this.lastSelectedUserEmail) {
             this.urlPointingUUID = newUser.uuid;
           }
         }
       }
     }
+
     this.renderingUsers = users;
     this.usersLoaded = true;
     this.dataLoaded = this.usersLoaded;
     if (this.dataLoaded) {
-      this.onDataLoaded()
+      this.onDataLoaded();
     }
   }
 
-  onDataLoaded() {
-    if (isNullOrUndefined(this.urlPointingUUID)) {
-      this.setCurrentUser(this.renderingUsers[0]);
+  private onDataLoaded() {
+    if (!this.urlPointingUUID) {
+      this.setCurrentUser({user: this.renderingUsers[0]});
     } else {
-      let foundUser = this.renderingUsers.find((val) => val.uuid == this.urlPointingUUID);
-      if (isNullOrUndefined(foundUser)) {
-        this.setCurrentUser(this.renderingUsers[0], false, true);
+      const foundUser = this.renderingUsers.find(
+          (user) => user.uuid === this.urlPointingUUID);
+      if (!foundUser) {
+        this.setCurrentUser({
+          user: this.renderingUsers[0],
+          fromInputChange: false,
+          shouldSetLastUser: true
+        });
       } else {
-        this.setCurrentUser(foundUser, false, true);
+        this.setCurrentUser({
+          user: foundUser,
+          fromInputChange: false,
+          shouldSetLastUser: true
+        });
       }
     }
   }
 
-  setCurrentUser(user: User, fromInputChange?: boolean, shouldSetLastUser?: boolean) {
+  setCurrentUser({user, fromInputChange, shouldSetLastUser}: {
+    user: User | undefined,
+    fromInputChange?: boolean,
+    shouldSetLastUser?: boolean,
+  }) {
```

Comment:
Is this to force the caller to specify parameters?

Reply:
Correct, this allows callers to be more readable as it's clear what parameters are being set. It also makes the arguments position-agnostic even though they are optional.